proto-mailbox/src/test/kotlin/actor/proto/mailbox/DefaultMailboxTest.kt

```diff
@@ -0,0 +1,102 @@
+        val mailbox = newUnboundedMailbox()
+        assertEquals(MailboxStatus.IDLE, (mailbox as DefaultMailbox).status())
+        val invoker = TestMailboxInvoker()
+        mailbox.registerHandlers(invoker, Dispatchers.SYNCHRONOUS_DISPATCHER)
+        mailbox.postUserMessage("TestUserMessage1")
+        mailbox.postUserMessage("TestUserMessage2")
+        mailbox.postSystemMessage(SomeSystemMessage(17))
+        mailbox.postSystemMessage(SomeSystemMessage(18))
+        mailbox.postUserMessage("TestUserMessage3")
+        mailbox.postSystemMessage(SomeSystemMessage(19))
+
+        assertEquals("TestUserMessage1", invoker.userMessages[0])
+        assertEquals("TestUserMessage2", invoker.userMessages[1])
+        assertEquals("TestUserMessage3", invoker.userMessages[2])
+
+        assertEquals(SomeSystemMessage(17), invoker.systemMessages[0])
+        assertEquals(SomeSystemMessage(18), invoker.systemMessages[1])
+        assertEquals(SomeSystemMessage(19), invoker.systemMessages[2])
+    }
+
+    @Test
+    fun `check suspension and resumption of user messages`() {
+        val mailbox = newUnboundedMailbox()
+        assertEquals(MailboxStatus.IDLE, (mailbox as DefaultMailbox).status())
+        val invoker = TestMailboxInvoker()
+        mailbox.registerHandlers(invoker, Dispatchers.SYNCHRONOUS_DISPATCHER)
+        mailbox.postUserMessage("TestUserMessage1")
+        mailbox.postUserMessage("TestUserMessage2")
+        mailbox.postSystemMessage(SomeSystemMessage(17))
+        mailbox.postSystemMessage(SomeSystemMessage(18))
+        mailbox.postSystemMessage(SuspendMailbox)
+        mailbox.postUserMessage("TestUserMessage3")
+        mailbox.postSystemMessage(SomeSystemMessage(19))
+        // user messages after suspension should not be delivered
+        assertEquals("TestUserMessage1", invoker.userMessages[0])
+        assertEquals("TestUserMessage2", invoker.userMessages[1])
+        assertEquals(2, invoker.userMessages.size)
+        // system messages after suspension should be delivered anyway
+        assertEquals(SomeSystemMessage(17), invoker.systemMessages[0])
+        assertEquals(SomeSystemMessage(18), invoker.systemMessages[1])
+        assertEquals(SuspendMailbox, invoker.systemMessages[2])
+        assertEquals(SomeSystemMessage(19), invoker.systemMessages[3])
+        assertEquals(4, invoker.systemMessages.size)
+        // user messages after resumption should be delivered
+        mailbox.postSystemMessage(ResumeMailbox)
+        assertEquals("TestUserMessage3", invoker.userMessages[2])
+        assertEquals(5, invoker.systemMessages.size)
+
+    }
+}
+
+private class TestMailboxInvoker : MessageInvoker {
+    var escalatedFailures: MutableList<Exception> = mutableListOf()
+    var systemMessages: MutableList<SystemMessage> = mutableListOf()
+    var userMessages: MutableList<Any> = mutableListOf()
+
+    override suspend fun invokeSystemMessage(msg: SystemMessage) {
+        systemMessages.add(msg)
```

Comment:
This is nice depending on if we're going write more test this way one idea could be to expose a mixed list so we can assert like
```
assertEquals([systemMessage1, userMessage1], invoker.allMessages)
```